HeterogeneousCore/CUDAServices/src/CUDAService.cc

```diff
@@ -98,15 +99,15 @@ namespace {
     cudaCheck(cudaGetDevice(&device));
     for(int i=0; i<numberOfDevices; ++i) {
       cudaCheck(cudaSetDevice(i));
-      preallocate<edm::cuda::device::unique_ptr>([&](size_t size, cuda::stream_t<>& stream) {
+      preallocate<cudautils::device::unique_ptr>([&](size_t size, cuda::stream_t<>& stream) {
           return cs.make_device_unique<char[]>(size, stream);
         }, bufferSizes);
     }
     cudaCheck(cudaSetDevice(device));
   }
 
   void hostPreallocate(CUDAService& cs, const std::vector<unsigned int>& bufferSizes) {
-    preallocate<edm::cuda::device::unique_ptr>([&](size_t size, cuda::stream_t<>& stream) {
+    preallocate<cudautils::host::unique_ptr>([&](size_t size, cuda::stream_t<>& stream) {
```

Comment:
> was the former wrong ?

Yes. This function preallocates the pinned host memory. It compiled earlier because technically `edm::cuda::device::unique_ptr` and `edm::cuda::host::unique_ptr` were the same types (which is changed in this PR in 0a9e54c448c6a6558bde98c93c494cc86f4eb815)